react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js

```diff
@@ -170,3 +172,45 @@ export const getConsentsFromAPIForSearch = (searchObj, user, appInfo) => {
         });
 };
 
+export const withdrawConsent = async (
+    clientId,
+    consentId,
+    user,
+    applicationName,
+    setMessage,
+    setWithdrawMessageIcon,
+    setWithdrawIconId,
+    setShow
+) => {
+    try {
+        const getRevokeUrl = (consentId, user) => {
+            const adminUrl = `${CONFIG.BACKEND_URL}/admin/revoke?consentID=${consentId}`;
+            const defaultUrl = `${CONFIG.BACKEND_URL}/admin/revoke?consentID=${consentId}&userID=${user.email}`;
+
+            return user.role === 'customerCareOfficer' ? adminUrl : defaultUrl;
+        };
+
+        const response = await axios.delete(getRevokeUrl(consentId, user), {
+            headers: {
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${Cookies.get(User.CONST.OB_SCP_ACC_TOKEN_P1)}`,
+                'x-wso2-client-id': clientId,
+                'x-fapi-financial-id': 'open-bank'
+            },
+        });
+        if (response.status === 204) {
+            setMessage(withdrawLang.withdrawModalSuccessMsg + applicationName);
+            setWithdrawMessageIcon(faCheckCircle);
+            setWithdrawIconId('withdrawSuccess');
+        } else {
+            setMessage(withdrawLang.withdrawModalFailMsg);
+            setWithdrawMessageIcon(faExclamationCircle);
+            setWithdrawIconId('withdrawFail');
+        }
```

Comment:
Shouldn't this function return a promise, similar to getConsentsFromAPI?

Reply:
changed the `revokeConsent` function to return a Promise and moved response handling logics to `WithdrawStep2.jsc` in [106de5c](https://github.com/wso2/financial-open-banking/pull/49/commits/106de5ca75429194a33fa5474cba8b5fbdc115ff)